File: KI304.Slobodian.Lab3/src/InternetAdapter.java
import java.util.Random;
/**
* Class <code>InternetAdapter</code> implements paper tray of printer
*/
public class InternetAdapter {
    enum ConnectionTypes{
        NOT_CONNECTED, WIFI, ETHERNET
    }

    Random rand = new Random();

    private ConnectionTypes connectionType;
    private boolean isConnected;
    private int connectionStrength;
    private String wifiSSID;
    private String wifiPassword;
    /**
    * Constructor
    */
    public InternetAdapter() {
        this.connectionType = ConnectionTypes.NOT_CONNECTED;
        this.isConnected = false;
        this.connectionStrength = 0;
        this.wifiSSID = "";
        this.wifiPassword = "";
    }
    /**
    * Constructor
    *@param <code>CType</code> index of the connection type
    *@param <code>IsC</code> bool, if connected is true
    *@param <code>CStrength</code> strength of the connection
    *@param <code>ssid</code> ssid, if WiFi
    *@param <code>pass</code> password, if WiFi
    */
    public InternetAdapter(int CType, boolean IsC, int CStrength, String ssid, String pass){
        this.connectionType = ConnectionTypes.values()[CType];
        this.isConnected = IsC;
        this.connectionStrength = CStrength;
        this.wifiSSID = ssid;
        this.wifiPassword = pass;
    }
    /**
	 * Method returns the connection type
	 * @return The connection type
	 */
    public ConnectionTypes checkConnectionType(){
        return connectionType;
    }
    /**
	 * Method returns if printer connected to the internet
	 * @return If printer connected to the internet
	 */
    public boolean checkIfConnected(){
        return isConnected;
    }
    /**
	 * Method returns strength of the connection
	 * @return The strength of connection to the internet
	 */
    public int checkConnectionStrength(){
        return connectionStrength;
    }
    /**
	 * Method returns ssid of the WiFi
	 * @return The ssid of the WiFi
	 */
    public String checkWifiSSID(){
        return wifiSSID;
    }
    /**
	 * Method returns password of the WiFi
	 * @return The password of the WiFi
	 */
    public String checkWifiPassword(){
        return wifiPassword;
    }

    /**
	 * Method connects internet adapter to the WiFi network
	 * @param ssid SSID
     * @param pass Password
     * @return consequences of your actions
	 */
    public String connectToWifi(String ssid, String pass){
        isConnected = true;
        connectionType = ConnectionTypes.WIFI;
        connectionStrength = rand.nextInt(101);
        wifiSSID = ssid;
        wifiPassword = pass;
        return "Your printer succesfully connected to the WiFi '"+wifiSSID+"' with password '"+wifiPassword+"' and strength of the signal of "+connectionStrength+" percent.";
    }
    /**
	 * Method connects internet adapter to the Ethernet cable
     * @return consequences of your actions
	 */
    public String connectToEthernet(){
        isConnected = true;
        connectionType = ConnectionTypes.ETHERNET;
        connectionStrength = 100;
        wifiSSID = "";
        wifiPassword = "";
        return "Your printer succesfully connected to the Ethernet cable with strength of the signal of "+connectionStrength+" percent. Nice choice!";
    }
    /**
	 * Method disconnects internet adapter from the internet
     * @return consequences of your actions
	 */
    public String disconnect(){
        connectionType = ConnectionTypes.NOT_CONNECTED;
        isConnected = false;
        connectionStrength = 0;
        wifiSSID = "";
        wifiPassword = "";
        return "You disconnected your printer from any sources of internet. Now it cannot be hacked!";
    }
}

File: src/internetadapter.py
from enum import Enum
import random

class InternetAdapter:
    class ConnectionTypes(Enum):
        NOT_CONNECTED = 0
        WIFI = 1
        ETHERNET = 2

    def __init__(self):
        self.connectionType = self.ConnectionTypes.NOT_CONNECTED
        self.isConnected = False
        self.connectionStrength = 0
        self.wifiSSID = ""
        self.wifiPassword = ""
        self.rand = random.Random()

    def __init__(self, CType, IsC, CStrength, ssid, passw):
        self.connectionType = self.ConnectionTypes(CType)
        self.isConnected = IsC
        self.connectionStrength = CStrength
        self.wifiSSID = ssid
        self.wifiPassword = passw
        self.rand = random.Random()

    def checkConnectionType(self):
        return str(self.connectionType)

    def checkIfConnected(self):
        return self.isConnected

    def checkConnectionStrength(self):
        return self.connectionStrength

    def checkWifiSSID(self):
        return self.wifiSSID

    def checkWifiPassword(self):
        return self.wifiPassword

    def connectToWifi(self, ssid, passw):
        self.isConnected = True
        self.connectionType = self.ConnectionTypes.WIFI
        self.connectionStrength = self.rand.randint(0, 100)
        self.wifiSSID = ssid
        self.wifiPassword = passw
        return f"Your printer successfully connected to the WiFi '{self.wifiSSID}' with password '{self.wifiPassword}' and a signal strength of {self.connectionStrength} percent."

    def connectToEthernet(self):
        self.isConnected = True
        self.connectionType = self.ConnectionTypes.ETHERNET
        self.connectionStrength = 100
        self.wifiSSID = ""
        self.wifiPassword = ""
        return f"Your printer successfully connected to the Ethernet cable with a signal strength of {self.connectionStrength} percent. Nice choice!"

    def disconnect(self):
        self.connectionType = self.ConnectionTypes.NOT_CONNECTED
        self.isConnected = False
        self.connectionStrength = 0
        self.wifiSSID = ""
        self.wifiPassword = ""
        return "You disconnected your printer from any sources of the internet. Now it cannot be hacked!"
